bloggies_frontend/src/RoutedPages/RegisterStatusPage/index.tsx

```diff
@@ -71,42 +73,55 @@ const PaymentButton = styled(Button)`
 
 function RegisterStatusPage() {
   const history = useHistory();
+  const dispatch = useDispatch()
+  const [text, setText] = useState('')
+
   // pulls membership status from redux store
   const checkStatus = useSelector(
     (st: CustomReduxState) => st.user.membership_status
   );
 
-  let text: string | null = null;
   // home buttone redirects user to newfeed page
   let homeButton = (
     <NewsFeedButton onClick={() => history.push("/")}>
       {"Take me back to the newsfeed!"}
     </NewsFeedButton>
   );
+
+
+
   // payment button directs user to stripe success form
   let paymentButton = (
-    <PaymentButton
-      onClick={() => history.push("/payment/form")}
+    <PaymentButton 
+    onClick={() => { 
+      history.push("/payment/form") 
+      dispatch(createCustomer())
```

Comment:
Yes I l know what you mean but it seems to be working fine, however the JS engine does run synchronously. I'll change the order because the action should be dispatched before going to another screen though the createCustomer action is async.